src/connectors/current-refinements/connectCurrentRefinements.ts

```diff
@@ -0,0 +1,320 @@
+  unmount?: Unmounter
+) => CurrentRefinementsWidgetFactory<TCurrentRefinementsWidgetParams>;
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'current-refinements',
+  connector: true,
+});
+
+const connectCurrentRefinements: CurrentRefinementsConnector = (
+  renderFn,
+  unmountFn = noop
+) => {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    if (
+      (widgetParams || ({} as any)).includedAttributes &&
+      (widgetParams || ({} as any)).excludedAttributes
+    ) {
+      throw new Error(
+        withUsage(
+          'The options `includedAttributes` and `excludedAttributes` cannot be used together.'
+        )
+      );
+    }
+
+    const {
+      includedAttributes,
+      excludedAttributes = ['query'],
+      transformItems = (items: Refinement[]) => items,
+    } = widgetParams || ({} as CurrentRefinementsConnectorParams);
+
+    return {
+      $$type: 'ais.currentRefinements',
+
+      init({ helper, createURL, instantSearchInstance }) {
+        const items = transformItems(
+          getFilteredRefinements({
+            results: {},
+            state: helper.state,
+            helper,
+            includedAttributes,
+            excludedAttributes,
+          })
+        );
+
+        renderFn(
+          {
+            items,
+            refine: refinement => clearRefinement(helper, refinement),
+            createURL: refinement =>
+              createURL(clearRefinementFromState(helper.state, refinement)),
+            instantSearchInstance,
+            widgetParams,
+          },
+          true
+        );
+      },
+
+      render({ results, helper, state, createURL, instantSearchInstance }) {
+        const items = transformItems(
+          getFilteredRefinements({
+            results,
+            state,
+            helper,
+            includedAttributes,
+            excludedAttributes,
+          })
+        );
+
+        renderFn(
+          {
+            items,
+            refine: refinement => clearRefinement(helper, refinement),
+            createURL: refinement =>
+              createURL(clearRefinementFromState(helper.state, refinement)),
+            instantSearchInstance,
+            widgetParams,
+          },
+          false
+        );
+      },
+
+      dispose() {
+        unmountFn();
+      },
+    };
+  };
+};
+
+function getFilteredRefinements({
+  results,
+  state,
+  helper,
+  includedAttributes,
+  excludedAttributes,
+}) {
+  const clearsQuery =
+    (includedAttributes || []).indexOf('query') !== -1 ||
+    (excludedAttributes || []).indexOf('query') === -1;
+
+  const filterFunction = includedAttributes
+    ? item => includedAttributes.indexOf(item.attributeName) !== -1
+    : item => excludedAttributes.indexOf(item.attributeName) === -1;
+
+  const items = getRefinements(results, state, clearsQuery)
+    .filter(filterFunction)
+    // @ts-ignore: @TODO the types don't match exactly, this will need to get refactored.
```

Comment:
This was already not completely accurate. We'll need to reiterate on the final types once the helper is fully typed anyway (wrong types are coming from `getRefinement.ts`).

Reply:
Fixed in 92d44525461948622e91bafac6b8b9b69666c74b.